src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.MarkdownContentBuilder.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.LanguageServer
+{
+    internal static partial class ProtocolConversions
+    {
+        private readonly struct MarkdownContentBuilder : IDisposable
+        {
+            private readonly ArrayBuilder<string> _linesBuilder;
+
+            public MarkdownContentBuilder()
+            {
+                _linesBuilder = ArrayBuilder<string>.GetInstance();
+            }
+
+            public void Append(string text)
```

Comment:
Could text every contain embedded newlines?

Reply:
Ideally it wouldn't be called like that, but I'm also not sure how to enforce that.